### PR TITLE
Add extendable controller & configuration for setup

### PR DESCRIPTION
**Pull request by KiKoS0**
I think someone with better spring boot knowledge can shape
a better interface to create the Inngest client + `CommHandler` but
this is already better than manually creating a singleton and
having everything in a random controller.

Note: It might be better to not have a second adapter package and
instead have these spring boot helpers as a feature variant. But
I'll see how it goes with this PR first: https://github.com/inngest/inngest-kt/pull/27

I'm also adding a way to build the Inngest client configuration
possibly through a Builder pattern targeted to the specific
environment that the user wants to use.

The current interface (which needs better code documentation) is that a consumer application would subclass the controller and configuration as follows:

##### ExampleController.java
```java
@RestController
@RequestMapping(value = "/api/inngest")
public class ExampleController extends InngestController {
}
```
##### ExampleConfiguration.java
```java
@Configuration
public class ExampleConfiguration extends InngestConfiguration {

    @Override
    public HashMap<String, InngestFunction> functions() {
        HashMap<String, InngestFunction> functions = new HashMap<>();
        ...
        return functions;
    }

    @Override
    public Inngest inngestClient() {
        return new Inngest("spring_example");
    }
}
```

### CODE REVIEW

**Review by darwin67 — APPROVED** [review]
This is fine for now.
The idea of #27 is that there will be a `serve` function that will handle all the routes, and also take the `Inngest` client as an input so the route handler will have all the information it needs.

Having some compilation error locally, so haven't push the latest code up yet.

**Comment by darwin67** [comment]
this is the latest code, and kotlin is not happy about something, but can't tell what it is.
https://github.com/inngest/inngest-kt/pull/27/files#diff-a53270cbe78c3842c55b0a20286bd726f41e9e697070455fb5f75c0fb73a4104R15

**Comment by KiKoS0** [comment]
oh it was the Kotlin paratrooped `it`. 😏

**Comment by darwin67** [comment]
yeah, I just figured it out. annoying

File: inngest-spring-boot-adapter/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+plugins {
+    `java-library`
+    id("io.spring.dependency-management") version "1.1.4"
+}
+
+group = "com.inngest"
+version = "0.0.1-SNAPSHOT"
```
Comment (albertchae) [inline inngest-spring-boot-adapter/build.gradle.kts:7]: probably best to version all the subpackages of `inngest-kt` together. I think there's something we can add in the root gradle file. I'll take a note of this for now

File: inngest-spring-boot-demo/src/main/java/com/inngest/springbootdemo/DemoConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package com.inngest.springbootdemo;
+
+
+import com.inngest.*;
+import com.inngest.springboot.InngestConfiguration;
+import kotlin.jvm.functions.Function2;
```
Comment (albertchae) [inline inngest-spring-boot-demo/src/main/java/com/inngest/springbootdemo/DemoConfiguration.java:6]: we already discussed this but probably ideal if we don't have to pull in a kotlin Function here and can instead use lambdas (available in Java 8) or something like https://docs.oracle.com/javase/8/docs/api/java/lang/Runnable.html

**Pull request merged**